xdsl/dialects/llvm.py

```diff
@@ -564,6 +564,57 @@ def print(self, printer: Printer):
         printer.print(self.arg.type, " to ", self.res.type)
 
 
+class IntegerConversionOpNNeg(IRDLOperation, ABC):
+    arg = operand_def(IntegerType)
+    res = result_def(IntegerType)
+    traits = traits_def(NoMemoryEffect())
+    non_neg = opt_prop_def(UnitAttr, prop_name="nonNeg")
+
+    def __init__(
+        self,
+        arg: SSAValue,
+        res_type: Attribute,
+        attributes: dict[str, Attribute] = {},
+        non_neg: UnitAttr | None = None,
+    ):
+        super().__init__(
+            operands=[arg],
+            attributes=attributes,
+            result_types=[res_type],
+            properties={
+                "nonNeg": non_neg,
+            },
+        )
+
+    @classmethod
+    def parse_nneg(cls, parser: Parser):
+        if parser.parse_optional_keyword("nneg") is not None:
+            return UnitAttr()
+
+    def print_nneg(self, printer: Printer) -> None:
+        if self.non_neg:
+            printer.print(" nneg")
+
+    @classmethod
+    def parse(cls, parser: Parser):
+        non_neg = cls.parse_nneg(parser)
+        arg = parser.parse_unresolved_operand()
+        attributes = parser.parse_optional_attr_dict()
+        parser.parse_characters(":")
+        arg_type = parser.parse_type()
+        parser.parse_characters("to")
+        res_type = parser.parse_type()
+        operands = parser.resolve_operands([arg], [arg_type], parser.pos)
+        return cls(operands[0], res_type, attributes, non_neg)
+
+    def print(self, printer: Printer):
+        self.print_nneg(printer)
+        printer.print(" ", self.arg)
+        printer.print_op_attributes(self.attributes)
+        printer.print(" : ")
+        printer.print(self.arg.type, " to ", self.res.type)
```

Comment:
Not sure how urgently you need this in, but could you try using `assembly_format` for this? I have a feeling it could work, and have all the functionality you need in one line instead of 25.

Reply:
Oh, I did not know that was a thing.
I'll try it, and make a note to work on using it for the other operations as well

Reply:
Done